yt_dlp/extractor/instagram.py

```diff
@@ -727,11 +738,15 @@ def _real_extract(self, url):
         highlights = traverse_obj(videos, (f'highlight:{story_id}', 'items'), (user_id, 'items'))
         info_data = []
         for highlight in highlights:
+            highlight['title'] = story_title
```

Comment:
Not sure I like this; for users with an output template of `%(title)s.%(ext)s` this would be a breaking change... our autogenerated generic titles include id

Reply:
Like I said above, the autogenerated ID will always be "Video by None" for non-highlight story because the `highlight` dict for it has no `user` or `owner` key.

We can copy it from `videos` object if you prefer. But then I don't think a overly complex traverse_obj makes sense, we probably should just separate logic for highlights and non-highlights.

Reply:
Maybe we should change "Video" wording, indeed. But do we want to keep the `by None`?

Reply:
See https://github.com/yt-dlp/yt-dlp/pull/12397/commits/d0c468ef9320f41a0a18b2ada336579f34a57e32: I now just copy the `user` into highlight entirely, which should be should be fine as the existing four keys are totally the same.